numpy/testing/_private/utils.py

```diff
@@ -788,16 +788,17 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
         # np.ma.masked, which is falsy).
         if cond != True:
             n_mismatch = reduced.size - reduced.sum(dtype=intp)
-            percent_mismatch = 100 * n_mismatch / ox.size
+            n_elements = reduced.size + flagged.sum(dtype=intp)
```

Comment:
Possibly clearer as the following, which also avoids the sum:
```python
# if we computed flags, then they are the full size of the comparison
n_elements = flagged.size if flagged.ndim != 0 else reduced.size
```

Lets see if anyone else agrees before changing.

Reply:
Yes, that's better. Thanks @eric-wieser